File: api-serena/src/main/java/com/taguz91/api_serena/models/Methodology.java
package com.taguz91.api_serena.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.io.Serial;
import java.io.Serializable;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Data
@Entity
@Table(name = "methodologies")
@Accessors(chain = true)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Methodology extends BaseEntity implements Serializable {
    @Serial
    private static final long serialVersionUID = -4115808525376597079L;

    @Id
    private String id;

    @Column
    private String name;

    @Column
    @Lob
    private String summary;

    @OneToMany(
            cascade = CascadeType.ALL,
            mappedBy = "methodology",
            fetch = FetchType.LAZY
    )
    private List<MethodologyEmotion> emotions;
}

File: web-serena/src/interfaces/methodology.ts
export interface Methodology {
  createdAt: string
  updatedAt: string
  createdBy: string
  updatedBy?: string
  id: string
  name: string
  summary: string
  emotions: MethodologyEmotion[]
}

export interface MethodologyEmotion {
  id: string
  emotion: string
}

export interface MethodologyForm {
  id?: string
  name: string
  summary: string
  emotions: string[]
}
